notifier/service/service.go

```diff
@@ -0,0 +1,276 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/jmoiron/sqlx"
+	"github.com/quay/clair/v4/indexer"
+	"github.com/quay/clair/v4/matcher"
+	"github.com/quay/clair/v4/notifier"
+	namqp "github.com/quay/clair/v4/notifier/amqp"
+	"github.com/quay/clair/v4/notifier/migrations"
+	"github.com/quay/clair/v4/notifier/postgres"
+	"github.com/quay/clair/v4/notifier/stomp"
+	"github.com/quay/clair/v4/notifier/webhook"
+	"github.com/quay/clair/v4/pkg/pager"
+	pgdl "github.com/quay/claircore/pkg/distlock/postgres"
+	"github.com/remind101/migrate"
+	"github.com/rs/zerolog"
+)
+
+const (
+	processors = 4
+	deliveries = 4
+)
+
+// Service is an interface wrapping ClairV4's notifier functionality.
+type Service interface {
+	// Retrieves an optional paginated set of notifications given an notification id
+	Notifications(ctx context.Context, id uuid.UUID, page *pager.Page) ([]notifier.Notification, pager.Page, error)
```

Comment:
`Pager.Page` should just be a part of the `notifier` package.

Reply:
That seems not right, this struct will be used for *any* paging api we use moving forward. the chances of it being shared and being moved to another package is high.